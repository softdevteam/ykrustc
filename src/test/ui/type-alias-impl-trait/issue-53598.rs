// ignore-compare-mode-chalk
// revisions: min_tait full_tait
#![feature(min_type_alias_impl_trait)]
#![cfg_attr(full_tait, feature(type_alias_impl_trait))]
//[full_tait]~^ WARN incomplete

use std::fmt::Debug;

pub trait Foo {
    type Item: Debug;

    fn foo<T: Debug>(_: T) -> Self::Item;
}

#[derive(Debug)]
pub struct S<T>(std::marker::PhantomData<T>);

pub struct S2;

impl Foo for S2 {
    type Item = impl Debug;

    fn foo<T: Debug>(_: T) -> Self::Item {
    //~^ Error type parameter `T` is part of concrete type but not used in parameter list for the `impl Trait` type alias
        S::<T>(Default::default())
    }
}

fn main() {
    S2::foo(123);
}
