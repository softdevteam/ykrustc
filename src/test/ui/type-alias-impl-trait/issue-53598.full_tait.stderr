warning: the feature `type_alias_impl_trait` is incomplete and may not be safe to use and/or cause compiler crashes
  --> $DIR/issue-53598.rs:4:32
   |
LL | #![cfg_attr(full_tait, feature(type_alias_impl_trait))]
   |                                ^^^^^^^^^^^^^^^^^^^^^
   |
   = note: `#[warn(incomplete_features)]` on by default
   = note: see issue #63063 <https://github.com/rust-lang/rust/issues/63063> for more information

error: type parameter `T` is part of concrete type but not used in parameter list for the `impl Trait` type alias
  --> $DIR/issue-53598.rs:23:42
   |
LL |       fn foo<T: Debug>(_: T) -> Self::Item {
   |  __________________________________________^
LL | |
LL | |         S::<T>(Default::default())
LL | |     }
   | |_____^

error: aborting due to previous error; 1 warning emitted

